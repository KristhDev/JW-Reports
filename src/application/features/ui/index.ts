export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';