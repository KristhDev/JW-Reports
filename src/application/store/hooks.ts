import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();