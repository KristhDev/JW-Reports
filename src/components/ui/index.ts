export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';