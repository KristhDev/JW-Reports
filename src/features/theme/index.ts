export { default as themeReducer, setTheme } from './slice';