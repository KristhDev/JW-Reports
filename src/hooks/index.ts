export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';