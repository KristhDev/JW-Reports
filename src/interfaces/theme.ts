export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}