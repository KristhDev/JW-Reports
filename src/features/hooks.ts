import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;