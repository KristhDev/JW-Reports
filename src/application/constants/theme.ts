import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];