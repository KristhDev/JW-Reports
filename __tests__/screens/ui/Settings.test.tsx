import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import hexToRgba from 'hex-to-rgba';
import dayjs from 'dayjs';

import { Settings } from '../../../src/screens/ui';

import { useStatus, useTheme } from '../../../src/hooks';

import { darkColors } from '../../../src/theme';

import { navigateMock } from '../../../jest.setup';

const setStatusMock = jest.fn();

jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <Settings /> screen', () => {
    (useStatus as jest.Mock).mockReturnValue({
        setStatus: setStatusMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors, theme: 'dark' },
        BUTTON_TRANSLUCENT_COLOR: hexToRgba(darkColors.button, 0.50),
        setTheme: jest.fn()
    });

    beforeEach(() => {
        render(<Settings />);
        jest.clearAllMocks();
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should call navigate of useNavigation when press buttons sections in MI CUENTA section', () => {
        const touchables = screen.getAllByTestId('section-btn-touchable');

        fireEvent.press(touchables[0]);

        expect(navigateMock).toBeCalledTimes(1);
        expect(navigateMock).toBeCalledWith('ProfileScreen');

        fireEvent.press(touchables[1]);

        expect(navigateMock).toBeCalledTimes(2);
        expect(navigateMock).toBeCalledWith('CredentialsScreen');
    });

    it('should call setStatus with respective values', () => {
        const touchables = screen.getAllByTestId('section-btn-touchable');

        fireEvent.press(touchables[6]);

        expect(setStatusMock).toHaveBeenCalledTimes(1);
        expect(setStatusMock).toHaveBeenCalledWith({
            code: 200,
            msg: 'Para más información o dejar sus comentarios acerca de la aplicación, escriba al correo: kristhdev@gmail.com',
        });
    });

    it('should render respective text in copyright', () => {
        const copyrightText = screen.getByTestId('settings-copyright-text');

        expect(copyrightText).toBeTruthy();
        expect(copyrightText.props.children.join('')).toBe(`Copyright © ${ dayjs().year() }`);
    });
});