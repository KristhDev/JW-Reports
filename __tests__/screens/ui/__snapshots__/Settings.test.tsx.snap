// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Settings /> screen should to match snapshot 1`] = `
Array [
  <RCTScrollView
    overScrollMode="never"
  >
    <View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#292929",
              "borderBottomWidth": 1,
              "paddingTop": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 10,
              },
              Object {
                "marginHorizontal": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 17,
                  "textAlign": "center",
                },
                Object {
                  "color": "#93A8AB",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
            testID="info-text-text"
          >
            MI CUENTA
          </Text>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Perfil
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              Actualice sus datos personales
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Credenciales
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              Cambie sus credenciales (correo y contraseña)
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#292929",
              "borderBottomWidth": 1,
              "paddingTop": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 10,
              },
              Object {
                "marginHorizontal": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 17,
                  "textAlign": "center",
                },
                Object {
                  "color": "#93A8AB",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
            testID="info-text-text"
          >
            PANTALLA
          </Text>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Apariencia
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              Modo oscuro
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#292929",
              "borderBottomWidth": 1,
              "paddingTop": 15,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 10,
              },
              Object {
                "marginHorizontal": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 17,
                  "textAlign": "center",
                },
                Object {
                  "color": "#93A8AB",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
            testID="info-text-text"
          >
            PRIVACIDAD
          </Text>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Permisos
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              Admita o rechace los permisos de la aplicación (tenga en cuenta que ciertas funcionalidades se verán afectadas).
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#292929",
              "borderBottomWidth": 1,
              "paddingTop": 15,
            },
            Object {
              "borderBottomWidth": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "padding": 10,
              },
              Object {
                "marginHorizontal": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 17,
                  "textAlign": "center",
                },
                Object {
                  "color": "#93A8AB",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
            testID="info-text-text"
          >
            SOBRE
          </Text>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Versión
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              1.0.0
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Repositorio
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              Código fuente de la aplicación
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 16,
              },
            ]
          }
          testID="section-btn-touchable"
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                }
              }
              testID="section-btn-text"
            >
              Más información
            </Text>
            <Text
              style={
                Object {
                  "color": "#BFBFBF",
                  "fontSize": 14,
                }
              }
              testID="section-btn-sub-text"
            >
              Obtenga información sobre la aplicación o deje sus comentarios
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#BFBFBF",
            "fontSize": 14,
            "padding": 20,
          }
        }
        testID="settings-copyright-text"
      >
        Copyright © 
        2023
      </Text>
    </View>
  </RCTScrollView>,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  />,
]
`;
